controllers/storagecluster/external_resources.go

```diff
@@ -299,6 +299,30 @@ func (obj *ocsExternalResources) ensureDeleted(r *StorageClusterReconciler, inst
 		}
 		defer externalClusterClient.Close()
 
+		if data, ok := externalOCSResources[instance.UID]; ok {
+			for _, d := range data {
+				switch d.Kind {
+				case "CephFilesystemSubVolumeGroup":
+					cephFilesystemSubVolumeGroup := &cephv1.CephFilesystemSubVolumeGroup{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      d.Name,
+							Namespace: instance.Namespace,
+						},
+					}
+					err = r.Client.Delete(context.TODO(), cephFilesystemSubVolumeGroup)
+					if err != nil && !errors.IsNotFound(err) {
+						r.Log.Error(
+							err,
+							"Uninstall: Failed to delete CephFilesystemSubVolumeGroup.",
+							"CephFilesystemSubVolumeGroup",
+							klog.KRef(cephFilesystemSubVolumeGroup.Namespace, cephFilesystemSubVolumeGroup.Name),
+						)
+						return reconcile.Result{}, fmt.Errorf("uninstall: Failed to delete CephFilesystemSubVolumeGroup %v: %v", cephFilesystemSubVolumeGroup.Name, err)
+					}
+				}
+			}
+		}
+
 		return r.offboardConsumer(instance, externalClusterClient)
```

Comment:
I think we should offBoard before deleting these resources. Can you move this offboard call before the deletion?

Reply:
Fixed

Reply:
Hi @Madhu-1 here is a code where we delete all resources https://github.com/red-hat-storage/ocs-operator/blob/main/controllers/storagecluster/uninstall_reconciler.go#L316